packages/frontend/src/pages/SavedSearches/useSavedSearches.ts

```diff
@@ -32,13 +53,81 @@ export const filterSavedSearches = (
 };
 
 export const useSavedSearches = (selectedPartyIds?: string[]): UseSavedSearchesOutput => {
+  const [isSavingSearch, setIsSavingSearch] = useState<boolean>(false);
+  const [isDeletingSearch, setIsDeletingSearch] = useState<boolean>(false);
```

Comment:
No need to separate "isSavingSearch" and "isDeletingSearch", simply exposing "isLoading" is enough. 
Then we avoid having extra logic in SaveSearchButton.tsx, we could just consume isLoading.